src/main/kotlin/graphics/scenery/geometry/curve/FrenetFrameCalculation.kt

```diff
@@ -0,0 +1,116 @@
+package graphics.scenery.geometry.curve
+
+import graphics.scenery.geometry.Spline
+import graphics.scenery.utils.extensions.toFloatArray
+import graphics.scenery.utils.lazyLogger
+import org.joml.AxisAngle4f
+import org.joml.Quaternionf
+import org.joml.Vector3f
+import kotlin.math.acos
+
+
+/**
+ * Computes the Frenet Frames along a curve defined by a spline.
+ * @param spline: spline along which the frenet frames are to be calculated
+ * @param [firstPerpendicularVector] vector to which the first frenet tangent shall be perpendicular to.
+ */
+object FrenetFrameCalculation{
```

Comment:
I feel this could be in `Curve`, not?

Reply:
I am not sure about this one. I don't think it should be in `Curve` because `Curve` is even abstracted from frenet frames (e.g., `Helix` is a Curve that requires no frenet frames to be computed). It could be part of `FrenetOrientedCurve` but if someone were to use a different way of computing the frames, the two components would be again fused.  On the other hand, this is unlikely. So if you agree, then lets go with your intuition!